quota/etcd/quotapb/quotapb.proto

```diff
@@ -133,6 +133,10 @@ message UpdateConfigRequest {
   // Fields modified by the update request.
   // For example: "state" or "max_tokens".
   google.protobuf.FieldMask update_mask = 3;
+
+  // If true the updated quota is reset, regardless of the updates's contents.
+  // A reset quota is replenished to its maximum number of tokens.
+  bool reset_quota = 4;
```

Comment:
It might give a 'leaner' API, but so would putting every operation into a single monster `DoSomething` request!

I can cope with the concept that a change to a quota's config might automatically reset the current state; I'm mostly struggling with the idea that (just) resetting a config is going to involve an `UpdateConfig` request, which is not where I'd think to look in a hurry (particularly given the contents of the `Config` struct don't mention anything relevant).  So it feels like it's been shoehorned in here to save the trouble of creating something separate, leading to odd special-case code like the stuff in quota_server.go:121-125 and the dangling `UpdateConfigRequest.reset_quota` field.

In terms of the [API Design Guide](https://cloud.google.com/apis/design/resource_names) having something against it, it seems like you're using the same resource name (`Config.name`) to refer to two different types of resource, `Config` and current state, and you're not implementing any of the [standard methods](https://cloud.google.com/apis/design/standard_methods) for the latter -- just a special-case `Update(value=0)`.

(Which does suggest a possible alternative -- if the current state were included in the `Config`, the semantics would be more straightforward (albeit with some constraints) and it would make things like `ListConfig` more useful by showing current state as well as limits.)

BTW, documentation on which API changes require/assume/allow changes to other API fields should live in the API definition, not in a separate README.